src/main/java/newbank/server/commands/MoveMoneyCommand.java

```diff
@@ -0,0 +1,69 @@
+package newbank.server.commands;
+
+import java.math.BigDecimal;
+import java.text.DecimalFormat;
+
+import org.javamoney.moneta.Money;
+
+import newbank.server.Account;
+import newbank.server.CustomerID;
+import newbank.server.NewBank;
+import newbank.server.exceptions.AccountBalanceInsufficientException;
+import newbank.server.exceptions.AccountInvalidException;
+import newbank.server.exceptions.CommandInvalidSyntaxException;
+import newbank.server.exceptions.RequestNotAllowedException;
+
+public class MoveMoneyCommand extends Command {
+  private final NewBank bank;
+  private final String[] tokens;
+  private final CustomerID customer;
+
+  public MoveMoneyCommand(final NewBank bank, final String[] tokens, final CustomerID customer) {
+    this.bank = bank;
+    this.tokens = tokens;
+    this.customer = customer;
+  }
+
+  public String getSyntax() {
+    return "MOVE <account_name_from> <account_name_to> <amount>";
+  }
+
+  @Override
+  public String execute() throws CommandInvalidSyntaxException {
+    try {
+      checkLoggedIn(customer);
+    } catch (RequestNotAllowedException ex) {
+      return ex.getMessage();
+    }
+
+    if (tokens.length != 4) {
+      throw new CommandInvalidSyntaxException();
+    }
+
+    final String accountNameFrom = tokens[1];
+    final String accountNameTo = tokens[2];
+    final String amountString = tokens[3];
+
+    if (accountNameFrom.equals(accountNameTo)) {
+      return String.format("FAIL: The accounts must be different to complete a transfer.");
+    }
+
+    try {
+      final BigDecimal amount = new BigDecimal(amountString);
```

Comment:
I'm able to move 0.001 GBP, which we'd presumably want to prevent. Should we be rejecting any input with greater than 2 decimal places..?
I think this is a problem more broadly though, so you could leave it and I'll fix it in one place on the microloans branch tomorrow morning.

Reply:
Ahh, that's no good. Indeed, we would need to check the string first.

Reply:
Raised separate item to tackle this. Closing.